pinot-core/src/main/java/org/apache/pinot/core/data/manager/BaseTableDataManager.java

```diff
@@ -329,7 +330,8 @@ public Map<String, SegmentErrorInfo> getSegmentErrors() {
   public void reloadSegment(String segmentName, IndexLoadingConfig indexLoadingConfig, SegmentZKMetadata zkMetadata,
       SegmentMetadata localMetadata, @Nullable Schema schema, boolean forceDownload)
       throws Exception {
-    File indexDir = getSegmentDataDir(segmentName);
+    String segmentTier = getSegmentCurrentTier(segmentName);
```

Comment:
We should store the tier name within the `localMetadata`. You should be able to set it within the `ImmutableSegmentLoader.load(SegmentDirectory segmentDirectory, IndexLoadingConfig indexLoadingConfig, @Nullable Schema schema)`

Reply:
I thought about using SegmentMetadataImpl at beginning, but I see its content is all from the metadata.properties file, so I assume it's used as an (effective) immutable object. And MutableSegment has SegmentMetadataImpl object too, but it doesn't have tier.

Reply:
Understood, but logically the tier info should be part of the metadata. E.g `indexDir` is not from the properties file. For the mutable one, it can simply return `null`.

Reply:
I'll refactor this in a separate PR, as it's orthogonal to the feature of this PR.